thoth-export-server/src/marc21/marc21record_thoth.rs

```diff
@@ -532,6 +553,47 @@ fn language_field(languages: &[WorkLanguages]) -> Option<FieldRepr> {
     Some(language_field)
 }
 
+fn toc_field(relations: &[WorkRelations]) -> ThothResult<FieldRepr> {
+    let mut chapters = relations
+        .iter()
+        .filter(|r| r.relation_type == RelationType::HAS_CHILD)
+        .collect::<Vec<_>>();
+    if chapters.is_empty() {
+        return Err(ThothError::IncompleteMetadataRecord(
+            MARC_ERROR.to_string(),
+            "No chapter data available for constructing contents note".to_string(),
+        ));
+    }
+    // WorkQuery should already have retrieved these sorted by ordinal, but sort again for safety
+    chapters.sort_by(|a, b| a.relation_ordinal.cmp(&b.relation_ordinal));
+    let mut toc_field: FieldRepr = FieldRepr::from((b"505", "00"));
+    let mut separator = " --";
+    for (i, chapter) in chapters.iter().enumerate() {
+        let chapter_title = &chapter.related_work.full_title;
+        let chapter_number = chapter.relation_ordinal;
+        let chapter_authors = chapter
+            .related_work
+            .contributions
+            .iter()
+            .map(|c| c.full_name.clone())
+            .collect::<Vec<_>>()
+            .join(", ");
+        toc_field = toc_field.add_subfield(b"g", format!("{}.", chapter_number))?;
+        if i == chapters.len() - 1 {
```

Comment:
Could use a [Peekable](https://doc.rust-lang.org/std/iter/struct.Peekable.html) instead, to avoid having counters